File: Seminar_1/quicksort/lib/math.c
#include <stdio.h>

/* Get median of three numbers (first, middle, last) by
 * performing XOR operation comparing values */
int medianthree(int *arr, int a, int b, int c)
{
	if ((arr[a] > arr[b]) ^ (arr[a] > arr[c])) 
		return a;
	else if ((arr[b] < arr[a]) ^ (arr[b] < arr[c])) 
		return b;
	else
		return c;
}

File: Python/Seminar_1/quicksort/lib/math.py
def median_three(arr: list, a: int, b: int, c: int):
        if ((arr[a] > arr[b]) ^ (arr[a] > arr[c])):
                return a
        elif ((arr[b] < arr[a]) ^ (arr[b] < arr[c])):
                return b
        else:
                return c